src/metabase/api/search.clj

```diff
@@ -0,0 +1,49 @@
+(ns metabase.api.search
+  (:require [compojure.core :refer [DELETE GET POST PUT]]
+            [honeysql.helpers :as h]
+            [metabase.util.honeysql-extensions :as hx]
+            [metabase.api.common :refer [defendpoint define-routes]]
+            [metabase.models
+             [dashboard :refer [Dashboard]]
+             [card :refer [Card]]
+             [collection :refer [Collection]]
+             [pulse :refer [Pulse]]]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(defn- create-search-query [entity entity-name search-string archived?]
```

Comment:
This is a little too cute, especially since we know we'll need to extend this for eg tables which have different names (eg "display_name"). Imo it's not worth saving a couple of lines of code in a couple places to have such a rigid expectation of which fields each type's search will hit.

Reply:
could always switch it to be a multimethod that dispatches off of entity down the road.

Reply:
so long as easily predicable things aren't made difficult -
just to write these out, things that are easily predictable:
1. we'll want to search more other model types
2. we'll want to experiment what is shown in the search listing for each of these types
3. we'll have slightly different ranking weights by type (explicitly or implicitly)

Reply:
FWIW, I think the queries will drift far enough apart soon that I won't be able to get away with what's there, just seemed wrong to start out with a bunch of copy and pasted code when it wasn't needed. In addition to the things you've described, I also think we'll need/want to use some specific query performance hacks for questions. I'm not worrying about it now as we're getting it working, but later versions of MySQL (5.6+) and PG have indexing and functions that would make these kinds of queries faster.

I expect it to be minor refactor to break it up and get more flexibility, just waiting to cross that threshold.

Reply:
cool. my main point is: "make the things that are easily predictable easy"